### PR TITLE
update modules to nixos 24.05 and k3s 1.30

### PR DESCRIPTION
**Pull request by yu-re-ka**

### CODE REVIEW

**Comment by bglgwyng** [comment]
I encountered this error message.
```
error:
       … while calling the 'head' builtin

         at /nix/store/0jr2kk95c34c0b6yxi75q4fqgb43kqkm-source/lib/attrsets.nix:1575:11:

         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'

         at /nix/store/0jr2kk95c34c0b6yxi75q4fqgb43kqkm-source/lib/modules.nix:821:9:

          820|     in warnDeprecation opt //
          821|       { value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          822|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute '"sha256-Mj9Q3TgqZoJluG4/nyuw2WHnB3OJ+/mlV7duzWt1B1A="' missing

       at /nix/store/3wy0ggfkfzma1hadbfy9b3jgn6272csz-source/modules/flake/overlays.nix:12:24:

           11|         buildGoModule = args: super.buildGoModule (args // super.lib.optionalAttrs (args.pname != "k3s-cni-plugins" && args.pname != "k3s-containerd") {
           12|           vendorHash = {
             |                        ^
           13|             "sha256-XtTahFaWnuHzKDI/U4d/j4C4gRxH163MCGEEM4hu/WM=" = "sha256-XuMP+ffwTdXKL9q9+ZJUQc5ghGEcdY9UdefjCD19OUE=";
```
Here is my [flake.lock](https://gist.github.com/bglgwyng/d1356786518ebc5b10b9acd939e5b7d6)

File: modules/flake/overlays.nix [hunk 1]
```diff
@@ -3,29 +3,20 @@
...
-        version = "1.7.14";
-        src = self.fetchFromGitHub {
-          owner = "containerd";
-          repo = "containerd";
-          rev = "v${version}";
-          hash = "sha256-okTz2UCF5LxOdtLDBy1pN2to6WHi+I0jtR67sn7Qrbk=";
-        };
-        patches = (o.patches or []) ++ [
-          # See: https://github.com/containerd/containerd/pull/9864
-          ./patches/containerd-import-compressed.patch
```
Comment (RobbieBuxton) [inline modules/flake/overlays.nix:18]: I think we still need to patch `containerd` here unless I'm mistaken. Although the PR was merged I don't think the [changes](https://github.com/containerd/containerd/pull/9864/commits/34c545824f5d49b771c13cc8e311bb3bec315fc0) have made their way into the version of [containerd currently in 24.05](https://github.com/NixOS/nixpkgs/blob/a781ff33ae258bbcfd4ed6e673860c3e923bf2cc/pkgs/applications/virtualization/containerd/default.nix#L14).

**Review by RobbieBuxton — CHANGES_REQUESTED** [review]
Hey @yu-re-ka,
Sorry for taking so long to get around to looking at this, it’s been a busy couple of months! Thanks again for the contributions!

File: modules/flake/overlays.nix (hunk 1 above)
Comment (RobbieBuxton) [reply modules/flake/overlays.nix:18]: Ah yes! My bad, no worries then.

**Review by RobbieBuxton — APPROVED** [review]
Ok I'm happy with this, if @elpdt852 is as well I think this is ready to merge!

**Comment by bglgwyng** [comment]
@yu-re-ka Thanks! your solution worked!
```
"sha256-Mj9Q3TgqZoJluG4/nyuw2WHnB3OJ+/mlV7duzWt1B1A=" = "sha256-9i0vY+CqrLDKYBZPooccX7OtFhS3//mpKTLntvPYDJo=";
```
I don't know well about `vendorHash` attribute in `buildGoModule`, but isn't it worth making PR to add this code? If then, I'll do it.

**Pull request merged**